org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/Pattern.java

```diff
@@ -39,6 +39,8 @@ public abstract class Pattern extends Expression {
 
 	public int nestingLevel = 0;
 
+	public int index = -1; // denoting position
+
 	@Override
```

Comment:
Comment can be more descriptive, As I look at it now, I am unable to conceive what this is for.